Streetcode/Streetcode.WebApi/Utils/WebParsingUtils.cs

```diff
@@ -247,158 +256,165 @@ await _repository.ToponymRepository.CreateAsync(new Toponym()
             Console.WriteLine("Success: " + (await _repository.SaveChangesAsync() > 0));
         }
 
-        // Following method returns name of the street optimized in such kind of way that will allow OSM Nominatim find its coordinates
-        public static Tuple<string?, string?> OptimizeStreetname(string streetname)
+        /// <summary>
+        /// Fetches the coordinates of an address using the OpenStreetMap Nominatim API.
+        /// </summary>
+        /// <param name="address">The address to fetch coordinates for.</param>
+        /// <returns>A tuple containing the latitude and longitude of the address.</returns>
+        public static async Task<(string, string)> FetchCoordsByAddressAsync(string address)
         {
-            var streetNameStringSearchOptimized = new Tuple<string?, string?>(null, null);
-
-            if (streetname.IndexOf("пров. ") != -1)
+            try
             {
-                streetNameStringSearchOptimized = new Tuple<string?, string?>(streetname.Substring(streetname.IndexOf(" ") + 1), "провулок");
+                using var client = new HttpClient();
+
+                // Add user-agent and referer headers to request
+                client.DefaultRequestHeaders.Add("user-agent", "Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.2; .NET CLR 1.0.3705;)");
+                client.DefaultRequestHeaders.Add("Referer", "http://www.microsoft.com");
+
+                // Send GET request to Nominatim API and retrieve JSON data
+                var jsonData = await client.GetStringAsync("https://nominatim.openstreetmap.org/search?q=" + address + "&format=json&limit=1&addressdetails=1");
```

Comment:
- With the current `HttpClient` setup, can you ensure that the `jsonData` will always be the string in `UTF-8` Encoding?
- No retry logic here?

Reply:
1 - done, 2 - we`ll add this later